src/app/credExplorer/weights/PluginWeightConfig.js

```diff
@@ -37,45 +37,43 @@ export class PluginWeightConfig extends React.Component<Props, State> {
   }
 
   _renderNodeWeightControls() {
-    return this.state.weights.nodes.map((wnt: WeightedNodeType) => {
-      const onChange = (newType: WeightedNodeType) => {
-        const index = this.state.weights.nodes.findIndex((x) =>
-          deepEqual(x.type, wnt.type)
+    return Array.from(this.state.weights.nodes.values()).map(
+      (wnt: WeightedNodeType) => {
+        const onChange = (newType: WeightedNodeType) => {
+          const newNodes = MapUtil.copy(this.state.weights.nodes);
+          newNodes.set(newType.type.prefix, newType);
+          const weights = {nodes: newNodes, edges: this.state.weights.edges};
```

Comment:
nit: prefer `{...this.state.weights, nodes: newNodes}`, unless a reason
otherwise? It’s a clearer signal to the reader that you intend to update
just one field.

Reply:
Done.